OpenRA.Game/Graphics/Util.cs

```diff
@@ -20,12 +20,27 @@ public static class Util
 		// yes, our channel order is nuts.
 		static readonly int[] ChannelMasks = { 2, 1, 0, 3 };
 
-		public static void FastCreateQuad(Vertex[] vertices, in float3 o, Sprite r, int2 samplers, float paletteTextureIndex, int nv, in float3 size, in float3 tint, float alpha)
+		public static void FastCreateQuad(Vertex[] vertices, in float3 o, Sprite r, int2 samplers, float paletteTextureIndex, int nv,
+			in float3 size, in float3 tint, float alpha, in float rotation = 0F)
```

Comment:
I would drop the `in` keyword when passing `float`. https://docs.microsoft.com/en-us/dotnet/csharp/write-safe-efficient-code#use-in-parameters-for-large-structs says "performance may degrade by using pass-by-reference for types smaller than [IntPtr.Size](https://docs.microsoft.com/en-us/dotnet/api/system.intptr.size#system-intptr-size)."

The `float3` parameters can keep it.

Reply:
Fixed